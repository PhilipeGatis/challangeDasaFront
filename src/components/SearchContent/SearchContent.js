import React from 'react';
import { Input, Row, Col } from 'antd';
import styled from 'styled-components';
import Lottie from 'react-lottie';
import animationData from './animation.json';

const { Search } = Input;

const StyledRow = styled(Row)`
  height: 100%;
`;

const StyledColInput = styled(Col)`
  height: 50%;
  display: flex;
  justify-content: center;
`;

const StyledSearch = styled(Search)`
  width: 50%;
  min-width: 350px;
`;

const StyledColLottie = styled(Col)`
  height: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const SearchContent = ({ history }) => {
  const handleSearch = (value) => {
    history.push(`/${value}`);
  };

  const defaultOptions = {
    loop: true,
    autoplay: true,
    animationData: animationData,
    rendererSettings: {
      preserveAspectRatio: 'xMidYMid slice',
    },
  };

  return (
    <StyledRow justify="center" align="middle">
      <StyledColLottie span={24}>
        <Lottie options={defaultOptions} height={200} width={200} />
      </StyledColLottie>
      <StyledColInput span={24}>
        <StyledSearch
          id="searchInput"
          placeholder="input username for search"
          enterButton="Search"
          size="large"
          onSearch={handleSearch}
        />
      </StyledColInput>
    </StyledRow>
  );
};

export default SearchContent;
