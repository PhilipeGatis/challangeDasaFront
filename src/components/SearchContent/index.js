import React from 'react';
import { Input } from 'antd';
import styled from 'styled-components';

const { Search } = Input;

const CenterContent = styled.section`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
`;

const SearchContent = ({ history }) => {
  const handleSearch = (value) => {
    history.push(`/${value}`);
  };

  const handleEnter = (evt) => {
    history.push(`/${evt.target.value}`);
  };

  return (
    <CenterContent>
      <Search
        placeholder="input search username"
        enterButton="Search"
        size="large"
        onPressEnter={handleEnter}
        onSearch={handleSearch}
      />
    </CenterContent>
  );
};

export default SearchContent;
