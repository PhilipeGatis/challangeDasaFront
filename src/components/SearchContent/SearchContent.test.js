import React from 'react';
import SearchContent from './SearchContent';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<SearchContent />', () => {
  it('should render', () => {
    const wrapper = render(<SearchContent/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
