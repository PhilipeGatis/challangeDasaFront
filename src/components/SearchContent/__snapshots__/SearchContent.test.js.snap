// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchContent /> should render 1`] = `
<div
  class="ant-row ant-row-center ant-row-middle sc-AxjAm kaRYdH"
>
  <div
    class="ant-col ant-col-24 sc-AxhCb eApdwK"
  >
    <div
      aria-label="animation"
      role="button"
      style="width:200px;height:200px;overflow:hidden;margin:0 auto;outline:none"
      tabindex="0"
      title=""
    />
  </div>
  <div
    class="ant-col ant-col-24 sc-AxirZ hcOHYQ"
  >
    <span
      class="ant-input-search sc-AxiKw ebfSXd ant-input-search-enter-button ant-input-search-large ant-input-group-wrapper ant-input-group-wrapper-lg"
    >
      <span
        class="ant-input-wrapper ant-input-group"
      >
        <input
          class="ant-input ant-input-lg"
          id="searchInput"
          placeholder="input username for search"
          type="text"
          value=""
        />
        <span
          class="ant-input-group-addon"
        >
          <button
            class="ant-btn ant-input-search-button ant-btn-primary ant-btn-lg"
            type="button"
          >
            <span>
              Search
            </span>
          </button>
        </span>
      </span>
    </span>
  </div>
</div>
`;
